lib/commands/network.js

```diff
@@ -58,7 +58,7 @@ commands.setNetworkConnection = async function (type) {
 
   if (shouldEnableWifi === isWiFiEnabled && shouldEnableDataConnection === isDataEnabled) {
     log.info('Not changing data connection/Wi-Fi states, since they are already set to expected values');
-    return;
+    return await this.getNetworkConnection();
```

Comment:
is there a reason we do the call to `getNetworkConnection` again? shouldn't we theoretically know what the state is now, without having to query it again?

Reply:
yep, this might be simplified

Reply:
I've checked it again and I think it would be better to keep it as it. Theoretically we might return just the expected connection state here and also at the end of the method, but we'd prefer to know if the connection state has really been changed. Also, the `getNetworkConnection` call itself is cheap

Reply:
ok, I've gotten some better solution.